vault/plugin_catalog.go

```diff
@@ -289,7 +296,7 @@ func (c *PluginCatalog) newPluginClient(ctx context.Context, pluginRunner *plugi
 
 	extPlugin.connections[id] = pc
 	extPlugin.name = pluginRunner.Name
-	extPlugin.multiplexingSupport = pluginRunner.MultiplexingSupport
+	extPlugin.multiplexingSupport = muxed
```

Comment:
This avoids having to return an `externalPlugin` that always sets multiplexingSupport to false if `pluginRunner.MultiplexingSupport` was set to false (e.g. on `PluginCatalog.Set`). Instead, we now query the plugin right after we have a clientConn. This allows us to pass the context ID properly if the plugin supports multiplexing so that [v5.PluginFactory](https://github.com/hashicorp/vault/blob/3565c90cf8ed54f27132821c20d58869b3f2856d/sdk/database/dbplugin/v5/plugin_factory.go#L15) can properly query `db.Type()` [in here](https://github.com/hashicorp/vault/blob/3565c90cf8ed54f27132821c20d58869b3f2856d/sdk/database/dbplugin/v5/plugin_factory.go#L67) for Vault to use a v5/v6 plugin.